Add more information about MathJax and LaTeX to the docs (#11819)
PrairieLearn/PrairieLearn: docs/question.md

<p>This is some <strong>Markdown</strong> text.</p>
```

!!! warning

>     It is reccommended to not indent the text inside markdown blocks, as markdown recognizes indentation as a code block.

Reviewer: Besides the typo (recommended, not reccommented), this sounds weird...
```suggestion
    Note that markdown recognizes indentation as a code block, so text inside these tags should not be indented with the corresponding HTML content.
```

Author: Oops, good catch. Thanks!

## Using LaTeX in questions

PrairieLearn supports LaTeX equations in questions. You can view a full list of supported MathJax commands [here](https://docs.mathjax.org/en/latest/input/tex/macros/index.html).

Inline equations can be written using `$x^2$`, and display equations can be written using `$$x^2$$` or `\[x^2\]`. For example:

```html title="question.html"
<p>Here is some inline math: $x^2$. Here is some display math: $$x^2$$</p>
<p>What is the total force $F$ currently acting on the particle?</p>
```

It can also be used inside `<markdown>` elements:

<!-- prettier-ignore -->
```html title="question.html"
<markdown>
  $$\phi = \frac{1+\sqrt{5}}{2}$$

  Need to use `$` outside of math-mode? Place inside inline code.

>   So, excel formulas would look like: `$A$1`, `B$5`, `$C9`, `D2`.

Reviewer: The issue with this is that it will change the `$` in different formatting, which is now always expected. You can also use `\$`, though this may be tricky in markdown (hopefully fixed by the followup to #10570 for server-side). Another option is to use `<span class="mathjax_ignore">$34.45</span>`. See https://prairielearn.readthedocs.io/en/latest/faq/#how-can-i-use-a-dollar-sign-without-triggering-math-mode.

>   So, excel formulas would look like: `$A$1`, `B$5`, `$C9`, `D2`.

Reviewer: Indentation in markdown may cause some issues, probably a good idea to avoid it here. Not sure if worth documenting as a reminder?

</pl-code>
```

## Using LaTeX in questions

PrairieLearn supports LaTeX equations in questions. You can view a full list of supported MathJax commands [here](https://docs.mathjax.org/en/latest/input/tex/macros/index.html).

> Inline equations can be written using `$x^2$`, and display equations can be written using `$$x^2$$` or `\[x^2\]`. For example:

Reviewer: ```suggestion
Inline equations can be written using `$x^2$` or `\(x^2\)`, and display equations can be written using `$$x^2$$` or `\[x^2\]`. For example:
```

</pl-code>
```

### Escaping `<markdown>` tags

Under the hood, PrairieLearn is doing some very simple parsing to determine what pieces of a question to process as Markdown: it finds an opening `<markdown>` tag and processes everything up to the closing `</markdown>` tag. But what if you want to have a literal `<markdown>` or `</markdown>` tag in your question? PrairieLearn defines a special escape syntax to enable this. If you have `<markdown#>` or `</markdown#>` in a Markdown block, they will be rendered as `<markdown>` and `</markdown>` respectively (but will not be used to find regions of text to process as Markdown). You can use more hashes to produce different strings: for instance, to have `<markdown###>` show up in the output, write `<markdown####>` in your question.

> ### Using a dollar sign ($) without triggering math mode

Reviewer 2: IMO this should be nested under the new section added below, not these docs about `<markdown>`.

Author: [03c7acb](https://github.com/PrairieLearn/PrairieLearn/pull/11819/commits/03c7acb7ada2775ace82abad511cbe2ec47d5f77)

Dollar signs by default denote either **inline** (`$ x $`) or **display mode** (`$$ x $$`) environments.

> To escape either math environment, consider using PrairieLearn's `markdown` tag and inline code syntax.

Reviewer 2: ```suggestion
To escape either math environment, consider using PrairieLearn's `<markdown>` tag and inline code syntax.
```

<!-- prettier-ignore -->
```html title="question.html"
<p>Here is some inline math: $x^2$. Here is some display math: $$x^2$$</p>
<p>What is the total force $F$ currently acting on the particle?</p>

<markdown>
# LaTeX works in Markdown too!

$$\phi = \frac{1+\sqrt{5}}{2}$$
</markdown>
```

### Using a dollar sign ($) without triggering math mode

Dollar signs by default denote either **inline** (`$ x $`) or **display mode** (`$$ x $$`) environments.

To escape either math environment, consider using PrairieLearn's `<markdown>` tag and inline code syntax.

<!-- prettier-ignore -->
```html
<markdown>
What happens if we use a `$` to reference the spreadsheet cell location `$A$1`?
</markdown>
```

> In scenarios that do not make sense for using the code environment, consider disabling math entirely by

Reviewer: ```suggestion
In scenarios where it does not make sense to use the code environment, consider disabling math entirely by
```
